drivers/serial/uart_ite_it8xxx2.c

```diff
@@ -0,0 +1,155 @@
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ite_it8xxx2_uart
+
+#include <device.h>
+#include <drivers/gpio.h>
+#include <drivers/uart.h>
+#include <kernel.h>
+#include <pm/device.h>
+#include <soc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(uart_ite_it8xxx2, LOG_LEVEL_ERR);
+
+#define DRV_CONFIG(dev) \
+	((const struct uart_it8xxx2_config *)(dev)->config)
+
+struct uart_it8xxx2_config {
+	/* UART handle */
+	const struct device *uart_dev;
+};
+
+#ifdef CONFIG_PM_DEVICE
+__weak void uart1_wui_isr_late(void) { }
+__weak void uart2_wui_isr_late(void) { }
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart1), okay)
+void uart1_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART1 RX pin to avoid repeated interrupts. */
+	gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+				     GPIO_INT_DISABLE);
+
+	if (uart1_wui_isr_late) {
+		uart1_wui_isr_late();
+	}
+}
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(uart1), okay) */
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay)
+void uart2_wui_isr(const struct device *gpio, struct gpio_callback *cb,
+		   uint32_t pins)
+{
+	/* Disable interrupts on UART2 RX pin to avoid repeated interrupts. */
+	gpio_pin_interrupt_configure(gpio, (find_msb_set(pins) - 1),
+				     GPIO_INT_DISABLE);
+
+	if (uart2_wui_isr_late) {
+		uart2_wui_isr_late();
+	}
+}
+#endif /* DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay) */
+
+static inline int uart_it8xxx2_pm_action_cb(const struct device *dev,
+					    enum pm_device_action action)
+{
+	const struct uart_it8xxx2_config *config = DRV_CONFIG(dev);
+	int ret;
+
+	switch (action) {
+	/* Next device power state is in active. */
+	case PM_DEVICE_ACTION_RESUME:
+		/* To do when EC is resuming. */
+		break;
+	/* Next device power state is deep doze mode */
+	case PM_DEVICE_ACTION_SUSPEND:
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart1), okay)
+		/* Enable UART1 WUI */
+		ret = gpio_pin_interrupt_configure(DEVICE_DT_GET(DT_NODELABEL(gpiob)),
+						   0, GPIO_INT_TRIG_LOW);
+		if (ret) {
+			LOG_ERR("Failed to configure UART1 WUI (ret %d)", ret);
+			return ret;
+		}
+#endif
+
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart2), okay)
+		/* Enable UART2 WUI */
+		ret = gpio_pin_interrupt_configure(DEVICE_DT_GET(DT_NODELABEL(gpioh)),
+						   1, GPIO_INT_TRIG_LOW);
+		if (ret) {
+			LOG_ERR("Failed to configure UART2 WUI (ret %d)", ret);
+			return ret;
+		}
+#endif
+		/*
+		 * If uart device is busy with transmitting, the driver will
+		 * stay in while loop and wait for the transaction is completed.
+		 */
+		while (!uart_irq_tx_complete(config->uart_dev)) {
+			continue;
+		}
+
+		break;
+	default:
+		return -ENOTSUP;
+	}
+
+	return 0;
+}
+#endif /* CONFIG_PM_DEVICE */
+
+static int uart_it8xxx2_init(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+	int ret;
+
+#ifdef CONFIG_PM_DEVICE
+#if DT_NODE_HAS_STATUS(DT_NODELABEL(uart1), okay)
```

Comment:
This block will get called twice, once for UART1 and again for UART2.  You should add properties to the ite,it8xxx2-uart node to specify the gpio-dev and port-num, similar to what you did for the I2C driver.

Then you should be able to remove all the `#if DT_NODE_HAS_STATUS(DT_NODELABEL(uartN), okay)` code blocks.

Reply:
Done.